Lib/contextlib.py

```diff
@@ -193,14 +193,6 @@ class _AsyncGeneratorContextManager(
 ):
     """Helper for @asynccontextmanager decorator."""
 
-    def __call__(self, func):
-        @wraps(func)
-        async def inner(*args, **kwds):
-            async with self.__class__(self.func, self.args, self.kwds):
```

Comment:
The inherited code is equivalent to `async with self`. I.e. it reuses the same object instead of recreating a new one.

If it is only used once or the context manager is stateless, there is no difference. Needed tests that exposes the difference.

Reply:
Ah, I see that `_recreate_cm()` has been overridden in `_GeneratorContextManagerBase`. If remove it, some tests start failing after applying this PR, as expected.